lib/src/network_request.dart

```diff
@@ -45,8 +49,17 @@ abstract class NetworkRequest {
   /// Query parameters for this request
   final Map<String, dynamic>? queryParams;
 
-  /// Headers for this request
-  final Map<String, String>? headers;
+  /// [Options] for this request
+  final Options? options;
+
+  /// [CancelToken] for this request
+  final CancelToken? cancelToken;
+
+  /// [ProgressCallback] for receive progress for this request
+  final ProgressCallback? onReceiveProgress;
+
+  /// [ProgressCallback] for send progress for this request
+  final ProgressCallback? onSendProgress;
```

Comment:
is this our first change that includes `Dio` library types in the public interfaces of this package?

It seems unlikely that we'd want to bother to avoid that coupling, but I figured I'd raise it as a question: is it a goal to keep the public API of this library from using other libraries' types outside the stdlib?

Reply:
> is this our first change that includes `Dio` library types in the public interfaces of this package?

No, since release (even internally) we've exposed the `Interceptor` type in the constructor. It had long been my hope that we'd actually remove all public `Dio` types from the public interface but I realized a while back that in order to do so there'd be a lot we'd essentially need to duplicate in order to do so. Since we prominently call out the coupling with `Dio` in the `README` I figured it'd be OK to further it. But I'm definitely open to alternative viewpoints. One extra note though: Unless I'm thinking about it wrong, if we wanted to get to the point where we didn't even bring in `Dio` as a transitive dependency to consumers, we'd likely need to provide a way for consumers to select their package of choice and provide their implementation via "adapters" (e.g `HttpClientAdapter`, `DioClientAdapter`, etc.) which would make things even more complex 🥴

Reply:
> general question about the motivation: did someone ask for this?

This work will directly support the migration of the [TT dart client](https://github.com/Betterment/test_track_dart_client) to `SturdyHttp`. It's actually just needed by our `RetryInterceptor`.

Reply:
yup, i'm in favor of just providing value on top of Dio. i think that's mostly always been the goal of this library, so this all seems reasonable to me. thanks for talking it out!